cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -370,7 +374,7 @@ public void performAction(FullHttpRequest request, HttpResponder responder,
   @POST
   @Path("/apps/{app-id}/versions/{app-version}/{program-type}/{program-id}/{action}")
   @AuditPolicy(AuditDetail.REQUEST_BODY)
-  public void performAction(FullHttpRequest request, HttpResponder responder,
+  public void performActionVersioned(FullHttpRequest request, HttpResponder responder,
```

Comment:
Don't need to do it in this PR, but it would have been better if we had explicit methods for each type and action (stopProgram, startProgram, suspendSchedule, resumeSchedule, etc.). Just pointing this out as an anti-pattern that we don't want to copy moving forward